### PR TITLE
feat: update advanced integration && update sdk

### PR DESCRIPTION
**Pull request by dmbsk**

### CODE REVIEW

File: src/dynamicToken/dynamicToken.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export const getTokenBaseOnTestNet = (
+  testNetName: EthTestNetsNames = getAppNetworkName()
+): string => {
+  if (process.env.REACT_APP_DEMO_APP_INSTANCE === "local") {
+    return fallbackToEnvDefault();
+  }
+  return TEST_NET_MAPPING_OBJECT[testNetName] ?? fallbackToEnvDefault();
+};
+console.log(getTokenBaseOnTestNet());
```
Comment (krszwsk) [inline src/dynamicToken/dynamicToken.ts:27]: console.log

File: src/dynamicToken/dynamicToken.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+const getDemoAppInstance = (): string =>
+  process.env.REACT_APP_DEMO_APP_INSTANCE!;
+
+const getAppNetworkName = () =>
+  process.env.REACT_APP_NETWORK_NAME?.toLocaleLowerCase() as EthTestNetsNames;
+
+const fallbackToEnvDefault = (): keyof typeof ENV_MAPPING_OBJECT => {
+  const fallbackToken = ENV_MAPPING_OBJECT[getDemoAppInstance()];
+  console.log(`Fallback to ${fallbackToken}`);
```
Comment (krszwsk) [inline src/dynamicToken/dynamicToken.ts:15]: console.log

File: src/dynamicToken/dynamicToken.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  return fallbackToken;
+};
+
+export const getTokenBaseOnTestNet = (
```
Comment (krszwsk) [inline src/dynamicToken/dynamicToken.ts:19]: Would `getTestNetTokenOrDefault` be clearer?

File: src/App.tsx [hunk 4]
```diff
@@ -3,14 +3,21 @@ import React, { useMemo, useState } from 'react';
 import './App.css';
```
Comment (krszwsk) [inline src/App.tsx:3]: Is max width applied in this file correct? Why wasn't it like this before?

File: src/App.tsx [hunk 5]
```diff
@@ -188,13 +216,33 @@ const App: React.FC = () => {
                   />
                   {currentNetwork === 'mainnet' ? '' : currentNetwork + ' '}ETH
                 </label>
-                <label className={styles.label} style={{ display: 'block' }} htmlFor="tokenRadio">
+                <label
+                  className={styles.label}
+                  style={{ display: 'block' }}
+                  htmlFor="ethRadio"
```
Comment (krszwsk) [inline src/App.tsx:222]: Should `ethRadio` be repeated here?

File: src/dynamicToken/dynamicToken.consts.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { EthMappingObject, EnvMappingObject } from "./dynamicToken.types";
+
+/*
+  How should be mapped
```
Comment (krszwsk) [inline src/dynamicToken/dynamicToken.consts.ts:4]:
```suggestion
  How it  should be mapped
```

**Review by krszwsk — CHANGES_REQUESTED** [review]

**Comment by krszwsk** [comment]
Closes #2269

File: src/dynamicToken/dynamicToken.ts (hunk 1 above)
Comment (dmbsk) [reply src/dynamicToken/dynamicToken.ts:27]: removed

File: src/App.tsx (hunk 5 above)
Comment (dmbsk) [reply src/App.tsx:222]: changed

File: src/dynamicToken/dynamicToken.ts (hunk 3 above)
Comment (dmbsk) [reply src/dynamicToken/dynamicToken.ts:19]: changed

File: src/dynamicToken/dynamicToken.consts.ts (hunk 6 above)
Comment (dmbsk) [reply src/dynamicToken/dynamicToken.consts.ts:4]: changed

File: src/dynamicToken/dynamicToken.ts (hunk 2 above)
Comment (dmbsk) [reply src/dynamicToken/dynamicToken.ts:15]: I wanted to notify the demo user about running, but now I think that is unnecessary.
removed

**Pull request merged**